tests/samples/test_tasks.py

```diff
@@ -215,3 +216,66 @@ async def test_deduplicate_sample_names(
     await task.run()
 
     assert await mongo.samples.find().to_list(None) == snapshot
+
+
+@pytest.mark.parametrize("ready", [[True, False], [False, False]])
+async def test_update_workflows_fields(
+    data_layer: DataLayer,
+    mongo,
+    pg: AsyncEngine,
+    ready,
+    static_time,
+    snapshot,
+):
+    samples = [
+        {
+            "_id": "test_id",
+            "library_type": "normal",
+            "nuvs": False,
+            "pathoscope": True,
+            "workflows": {
+                "aodp": "incompatible",
+                "nuvs": "none",
+                "pathoscope": "none",
+            },
+        },
+    ]
+    ready_1, ready_2 = ready
+    analyses = [
+        {
+            "_id": "test",
+            "sample": {"id": "test_id"},
+            "ready": ready_1,
+            "workflow": "pathoscope_bowtie",
+        },
+        {
+            "_id": "test1",
+            "sample": {"id": "test_id"},
+            "ready": ready_2,
+            "workflow": "nuvs",
+        },
+    ]
+
+    await mongo.samples.insert_many(samples, session=None)
+    await mongo.analyses.insert_many(analyses, session=None)
```

Comment:
I'd prefer you just put the `list` definitions right in the function call instead of defining `samples` and `analyses` separately.

Use whitespace between the `insert_many` calls to keep things legible.